### PR TITLE
osutils.py rewrite is_android() using get_android_api_version()

### PR DESCRIPTION
**Pull request by brussee**
Set proper paths on devices running Android and fetch the run-time API version.

### CODE REVIEW

**Comment by devos50** [comment]
add to whitelist

File: Tribler/Test/Core/test_osutils.py [hunk 1]
```diff
@@ -61,14 +61,21 @@ def test_fix_filebasename(self):
             fixedname = fix_filebasename(name)
             assert fixedname == name_table[name], (fixedname, name_table[name])
 
+    def test_get_android_api_version(self):
+        if sys.platform in ['win32','cygwin','darwin','os2','os2emx','riscos','atheos']:
+            self.assertIsNone(get_android_api_version())
+        elif 'ANDROID_PRIVATE' in os.environ:
```
Comment (lfdversluis) [inline Tribler/Test/Core/test_osutils.py]: you can get rid of one branch here by moving the `elif android_private in os.environ` to the first branch, and make the else case assertIsNone

File: Tribler/Test/Core/test_osutils.py [hunk 2]
```diff
@@ -61,14 +61,21 @@ def test_fix_filebasename(self):
...
+
     def test_is_android(self):
-        # act like the computer is an Android
-        self.assertFalse(is_android())
-        os.environ["ANDROID_HOST"] = "ANDROID_HOST"
-        self.assertTrue(is_android())
-        self.assertTrue(is_android(strict=True))
-        os.environ["ANDROID_HOST"] = "HOST"
-        self.assertFalse(is_android(strict=True))
+        if sys.platform in ['win32','cygwin','darwin','os2','os2emx','riscos','atheos']:
```
Comment (lfdversluis) [inline Tribler/Test/Core/test_osutils.py]: idem

**Comment by whirm** [comment]
We need an android test runner :)

**Comment by brussee** [comment]
@whirm: Yep. Do you want to merge this after that has been set up? This PR is quite ideal to verify the coverage tests.

**Comment by whirm** [comment]
Are you planning on setting up an android test runner?

**Comment by brussee** [comment]
As discussed an Android test runner would be significant slower than the other test runners. Also multiple platforms besides arm: x86 mips and 64bit of each etc would need their own build and test runner in the future.
So I propose to merge this PR now because as far as I can foresee now there won't be any more Android specific code needed in Tribler.Core and the configuration refactor can take care of the remaining issues on this topic: see #2139

**Comment by whirm** [comment]
Squash the commits and I'll merge

**Comment by whirm** [comment]
Let me know when it's ready

**Comment by lfdversluis** [comment]
or add the READY: tag in front of the title :D

**Comment by whirm** [comment]
retest this please

**Pull request merged**